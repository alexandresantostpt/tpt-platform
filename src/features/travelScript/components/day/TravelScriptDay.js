import { formats } from '@constants/dateFormats'

import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

import { i18n } from '@i18n'

import { navigateTo } from '@utils/browser'

import CitySelect from '../select/CitySelect'
import ServiceSelect from '../select/ServiceSelect'
import TravelScriptService from '../service/TravelScriptService'

const StyledTravelScriptDay = styled.div`
    margin: 0 2.062rem 3.75rem;
`

const StyledSelects = styled.div`
    display: flex;
    justify-content: space-between;
`

const StyledDate = styled.div`
    font-size: 0.85rem;
    font-weight: 300;
    margin-left: 2.56rem;
`

const StyledText = styled.div`
    font-size: 1rem;
    font-weight: 300;
    margin-top: 3.75rem;
    text-align: center;
`

const TravelScriptDay = ({
    cities,
    date,
    dateId,
    dayNumber,
    delService,
    id: travelId,
    scriptId,
    onCityChange,
    selectedCity,
    services
}) => {
    const serviceTip = services.find(service => service.isTip())
    const onServiceSelect = option => {
        const { value } = option
        const serviceRoute = value.replace(':id', travelId)
        navigateTo(
            `${serviceRoute}?scriptId=${scriptId}&scriptDate=${date.toISOString()}&dateId=${dateId}${
                selectedCity ? `&city=${selectedCity.name}&cityId=${selectedCity.id}` : ''
            }${serviceTip ? `&tipId=${serviceTip.id}` : ''}`
        )
    }

    const renderServices = () => {
        const dayServices = []
        if (services && services.length) {
            services.map((service, index) =>
                dayServices.push(
                    <TravelScriptService
                        dateId={dateId}
                        delService={delService}
                        key={index}
                        scriptDate={date.toISOString()}
                        scriptId={scriptId}
                        service={service}
                        travelId={travelId}
                    />
                )
            )
        } else {
            dayServices.push(<StyledText key="no-service">{i18n.t('messages.noServices')}</StyledText>)
        }

        return dayServices
    }

    return (
        <>
            <StyledTravelScriptDay>
                <StyledSelects>
                    <CitySelect
                        cities={cities}
                        dayNumber={dayNumber}
                        onSelect={option => onCityChange(option)}
                        selectedCity={selectedCity}
                    />
                    <ServiceSelect onSelect={onServiceSelect} />
                </StyledSelects>
                <StyledDate>{date.format(formats.shortest)}</StyledDate>
                {renderServices()}
            </StyledTravelScriptDay>
        </>
    )
}

TravelScriptDay.defaultProps = {
    selectedCity: null,
    services: []
}

TravelScriptDay.propTypes = {
    cities: PropTypes.arrayOf(PropTypes.object).isRequired,
    date: PropTypes.object.isRequired,
    dateId: PropTypes.string.isRequired,
    dayNumber: PropTypes.number.isRequired,
    id: PropTypes.string.isRequired,
    onCityChange: PropTypes.func.isRequired,
    scriptId: PropTypes.string.isRequired,
    selectedCity: PropTypes.object,
    services: PropTypes.array
}

export default TravelScriptDay
