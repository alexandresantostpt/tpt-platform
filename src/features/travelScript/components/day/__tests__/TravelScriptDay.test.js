import moment from 'moment'
import React from 'react'
import renderer from 'react-test-renderer'

import TravelScriptDay from '../TravelScriptDay'

describe('Tests for TravelScript component', () => {
    it('Should render withou crash', () => {
        const cities = ['Campinas', 'SÃ£o Paulo', 'Minas Gerais']
        const tree = renderer
            .create(
                <TravelScriptDay
                    cities={cities}
                    date={moment('2019-03-18').utc(false)}
                    dayNumber={1}
                    id={1}
                    onCityChange={jest.fn()}
                    selectedCity={'Campinas'}
                />
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
