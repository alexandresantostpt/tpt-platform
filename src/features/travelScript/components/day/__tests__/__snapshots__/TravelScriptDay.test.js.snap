// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for TravelScript component Should render withou crash 1`] = `
<div
  className="TravelScriptDay__StyledTravelScriptDay-uatujh-0 faduyg"
>
  <div
    className="TravelScriptDay__StyledSelects-uatujh-1 dFLRG"
  >
    <div
      className="CitySelect__StyledCitySelect-p43eaz-1 jykjhD"
    >
      <div
        className="CitySelect__StyledDayNumber-p43eaz-0 bsIoHV"
      >
        1
      </div>
      <div
        className="Select__StyledSelect-sc-12397s2-0 eBEwLw"
      >
        <div
          className="SelectContainer__StyledSelectContainer-an3t0v-0 jYoLEQ"
        >
          <span
            className="SelectText-eq3jk4-0 bQbIba"
            onClick={[Function]}
          >
            Selecione uma cidade...
          </span>
          <button
            className="SelectDropdown__StyledSelectDropdown-sc-12m970x-0 fwdrhv"
            onClick={[Function]}
            type="button"
          />
        </div>
        <ul
          className="SelectList__StyledSelectList-lfsn73-0 BSOrt"
        />
      </div>
      <div
        className="CitySelect__StyledLine-p43eaz-2 fDfKbI"
      />
    </div>
    <div
      className="ServiceSelect__StyledServiceSelect-sc-1rmby0e-0 kwWXAN"
    >
      <div
        className="Select__StyledSelect-sc-12397s2-0 eBEwLw"
      >
        <div
          className="SelectContainer__StyledSelectContainer-an3t0v-0 jYoLEQ"
        >
          <span
            className="SelectText-eq3jk4-0 bQbIba"
            onClick={[Function]}
          >
            <span
              className="icon icon-plus Icon__StyledIcon-crr1cg-0 cmjVJQ"
              disabled={false}
              fontSize="1.75rem"
              onClick={null}
            />
          </span>
          <button
            className="SelectDropdown__StyledSelectDropdown-sc-12m970x-0 fwdrhv"
            onClick={[Function]}
            type="button"
          />
        </div>
        <ul
          className="SelectList__StyledSelectList-lfsn73-0 BSOrt"
        />
      </div>
    </div>
  </div>
  <div
    className="TravelScriptDay__StyledDate-uatujh-2 kemUYJ"
  >
    18/03
  </div>
  <div
    className="TravelScriptDay__StyledText-uatujh-3 cAuILL"
  >
    Nenhum servi√ßo foi adicionado nesse dia
  </div>
</div>
`;
